game_mechanics.py

```diff
@@ -1,71 +1,449 @@
+    moves = get_legal_moves(board)
+    if moves:
+        return random.choice(moves)
+    return None
+
+
+####### THESE FUNCTIONS ARE LESS USEFUL ############
+
+
+def _make_move(board: np.ndarray, move: Tuple[int, int], current_player: int) -> np.ndarray:
+    """Returns board after move has been made."""
+    if is_legal_move(board, move, current_player):
+        board_after_move = copy.deepcopy(board)
+        board_after_move[move] = current_player
+        board_after_move = flip_tiles(board_after_move, move, current_player)
+    else:
+        raise ValueError(f"Move {move} is not a valid move!")
+    return board_after_move
+
+
+def is_legal_move(board: np.ndarray, move: Tuple[int, int], current_player: int) -> bool:
+    """Method: is_legal_move
+    Parameters: self, move (tuple)
+    Returns: boolean (True if move is legal, False otherwise)
+    Does: Checks whether the player's move is legal.
+    """
+    board_dim = board.shape[0]
+    if is_valid_coord(board_dim, move[0], move[1]) and board[move] == 0:
+        for direction in MOVE_DIRS:
+            if has_tile_to_flip(board, move, direction, current_player):
+                return True
+    return False
+
+
+def is_valid_coord(board_dim: int, row: int, col: int) -> bool:
+    """Method: is_valid_coord
+    Parameters: self, row (integer), col (integer)
+    Returns: boolean (True if row and col is valid, False otherwise)
+    Does: Checks whether the given coordinate (row, col) is valid.
+            A valid coordinate must be in the range of the board.
+    """
+    return 0 <= row < board_dim and 0 <= col < board_dim
+
+
+def has_tile_to_flip(
+    board: np.ndarray,
+    move: Tuple[int, int],
+    direction: Tuple[int, int],
+    current_player: int,
+) -> bool:
+    """Method: has_tile_to_flip
+    Parameters: self, move (tuple), direction (tuple)
+    Returns: boolean
+                (True if there is any tile to flip, False otherwise)
+    Does: Checks whether the player has any adversary's tile to flip
+            with the move they make.
+
+            About input: move is the (row, col) coordinate of where the
+            player makes a move; direction is the direction in which the
+            adversary's tile is to be flipped (direction is any tuple
+            defined in MOVE_DIRS) -> None.
+    """
+
+    board_dim = board.shape[0]
+    i = 1
+    while True:
+        row = move[0] + direction[0] * i
+        col = move[1] + direction[1] * i
+        if not is_valid_coord(board_dim, row, col) or board[row, col] == 0:
+            return False
+        elif board[row, col] == current_player:
+            break
+        else:
+            i += 1
+
+    return i > 1
+
+
+def flip_tiles(board: np.ndarray, move: Tuple[int, int], current_player: int) -> np.ndarray:
+    """Flips the adversary's tiles for current move and updates the running tile count for each
+    player.
+
+    Arg:
+        move (Tuple[int, int]): The move just made to
```

Comment:
I'd cut types from the docstring in general. It's repeated information and therefore unnecessary. Also means this line doesn't need to wrap